src/tribler/core/components/component.py

```diff
@@ -47,8 +50,16 @@ async def start(self):
             self._set_component_status(msg, logging.ERROR, exc_info=exc_info)
             self.failed = True
             self.started_event.set()
+
+            if isinstance(e, DatabaseIsCorrupted):
+                # When the database corruption is detected, we should stop the process immediately.
+                # Tribler GUI will restart the process and the database will be recreated.
+                process_manager = get_global_process_manager()
+                process_manager.sys_exit(EXITCODE_DATABASE_IS_CORRUPTED, e)
```

Comment:
As it is essential logic of the PR it should be tested.

Reply:
Added the test